spec/support/fake_cronitor.rb

```diff
@@ -12,18 +16,24 @@ class FakeCronitor < Sinatra::Base
   post '/v1/monitors' do
     payload = JSON.parse request.body.read
     # Check that we have the necessary payload values very simply
-    %w(name rules notifications).each do |k|
+    %w[name rules notifications].each do |k|
       return json_response 400, "invalid_no_#{k}" unless payload.key? k
     end
     json_response 200, 'new_monitor'
   end
 
-  %w(run complete fail).each do |ping|
+  %w[run complete fail].each do |ping|
     get "/abcd/#{ping}" do
-      200
+      content_type :json
+      status 200
+      params.to_json
     end
   end
 
+  before do
+    self.class.requests = [self.class.requests, request]
+  end
+
```

Comment:
Ah ok. Would you mind adding a code comment to that effect? The code itself is a little...opaque.